### PR TITLE
[dev] Add linters and run to all files

### PR DESCRIPTION
**Pull request by pdonorio**
Clean up the repo a bit
and setup the pre-commit hook to also use
linters like `black` and `isort`.

### CODE REVIEW

File: opyn/opyn/encode.py [hunk 1]
```diff
@@ -41,349 +37,355 @@
...
+        self.primaryType = primaryTypes[0]
+
+        def checkCircular(type: str, found: dict):
+            if type in found.keys():
+                raise ValueError(f'Circular type reference to {type}')
+
+            found[type] = True
+
+            for child in self.links[type].keys():
+                if child not in self.parents.keys():
```
Comment (mdantonio) [inline opyn/opyn/encode.py:184]: .keys() is not needed here

File: opyn/opyn/encode.py [hunk 2]
```diff
@@ -41,349 +37,355 @@
...
+
+        self.primaryType = primaryTypes[0]
+
+        def checkCircular(type: str, found: dict):
+            if type in found.keys():
+                raise ValueError(f'Circular type reference to {type}')
+
+            found[type] = True
+
+            for child in self.links[type].keys():
```
Comment (mdantonio) [inline opyn/opyn/encode.py:183]: .keys() is not needed here

File: opyn/opyn/encode.py [hunk 3]
```diff
@@ -41,349 +37,355 @@
...
+                if child not in self.parents.keys():
+                    checkCircular(child, found)
+
+                    for subtype in found.keys():
```
Comment (mdantonio) [inline opyn/opyn/encode.py:187]: Remove .keys() also here

File: opyn/opyn/encode.py [hunk 4]
```diff
@@ -41,349 +37,355 @@
...
+                        self.subtypes[subtype][child] = True
+
+            del found[type]
+
+        checkCircular(self.primaryType, {})
+
+        for name in self.subtypes.keys():
```
Comment (mdantonio) [inline opyn/opyn/encode.py:194]: remove .keys()

File: opyn/opyn/encode.py [hunk 5]
```diff
@@ -41,349 +37,355 @@
...
+            st = list(self.subtypes[name].keys())
+            st.sort()
```
Comment (mdantonio) [inline opyn/opyn/encode.py:196]: equivalent to:

```
st = sorted(self.subtypes[name].keys())
```

File: opyn/opyn/encode.py [hunk 6]
```diff
@@ -41,349 +37,355 @@
...
+            self._types[name] = encode_type(name, types[name]) + ''.join(
+                [encode_type(t, types[t]) for t in st]
+            )
+
+    def _get_encoder(self, type: str) -> object:
```
Comment (mdantonio) [inline opyn/opyn/encode.py:201]: build-in type shadowing

File: opyn/opyn/encode.py [hunk 7]
```diff
@@ -41,349 +37,355 @@
...
+
+        Args:
+            type (str): Data type in string
+
+        Returns:
+            encoder (object): Encoder
+        """
+        encoder = get_base_encoder(type)
+        if encoder is not None:
+            return encoder
```
Comment (mdantonio) [inline opyn/opyn/encode.py:213]:
```
if encoder := get_base_encoder(type):
    return encoder
```

File: opyn/opyn/encode.py [hunk 8]
```diff
@@ -41,349 +37,355 @@
...
+
+        Returns:
+            encoder (object): Encoder
+        """
+        if type in self._encoderCache.keys():
+            return self._encoderCache[type]
+        else:
+            encoder = self._get_encoder(type)
+
+        return encoder
```
Comment (mdantonio) [inline opyn/opyn/encode.py:263]:
```suggestion
        if type in self._encoderCache:
            return self._encoderCache[type]
        else:
            return self._get_encoder(type)
```

File: opyn/opyn/encode.py [hunk 9]
```diff
@@ -41,349 +37,355 @@
...
+        Encode the domain dictionary
+
+        Args:
+            domain (dict): Domain values in dictionary
+
+        Returns:
+            hash (str): Hash of encoded domain data
+        """
+        domainFields = []
+        for name in domain.keys():
```
Comment (mdantonio) [inline opyn/opyn/encode.py:343]: remove .keys

File: opyn/opyn/utils.py [hunk 10]
```diff
@@ -11,123 +11,131 @@
...
+    Args:
+        items (str): List of hexes with 0x prefix
+
+    Returns:
+        hex (str): Concatenated hex
+    """
+    result = '0x'
+    for i in items:
+        result += i[2:]
+    return result
```
Comment (mdantonio) [inline opyn/opyn/utils.py:69]: I'm not a big fan of one liner, but in that case could be considered:

```
return '0x' + ''.join([i[2:] for i in items])
```

**Review by mdantonio — APPROVED** [review]
I started to suggest some changes but since this PR only applies linters these suggestions are out of scope here and we can postpone them in a later step

I was tempted to undo all the comments, but it think that can be useful as a future reference. Feel free to ignore!

**Comment by pdonorio** [comment]
Forced push after rebasing

**Pull request merged**